sdks/python/apache_beam/testing/benchmarks/nexmark/nexmark_launcher.py

```diff
@@ -168,32 +189,45 @@ def run_query(query, args, pipeline_options):
       result.cancel()
     else:
       result.wait_until_finish()
-  except NotImplementedError as exp:
-    print(exp)
+  except Exception as exc:
+    thread_errors.append(str(exc))
 
 
 def run(argv=None):
   args, _ = parse_args()
 
   queries = {
       0: query0,
-      # TODO(mariagh): Add more queries
+      1: query1,
+      2: query2,
+      # TODO(mariagh): query4
   }
 
   for i in args.query:
     args, pipeline_options = parse_args()
     logging.info('Running query %d', i)
 
-    job_setup_time = 20 * 1000    # in milliseconds
-    query_duration = pipeline_options.view_as(TestOptions).wait_until_finish_duration + job_setup_time # pylint: disable=line-too-long
-    command = Command(run_query, args=[queries[i], args, pipeline_options])
-    try:
-      command.run(timeout=query_duration/1000)
-    except Exception as exp:
-      logging.error('Command failed with "%s"', exp)
-
-  logging.info('Queries run successfully: %s', args.query)
-
+    # The DirectRunner is the default runner, and it needs
+    # special handling to cancel streaming jobs.
+    launch_from_direct_runner = pipeline_options.view_as(
+        StandardOptions).runner in [None, 'DirectRunner']
+
+    if launch_from_direct_runner:
+      thread_errors = collections.defaultdict(list)
```

Comment:
Did you mean `thread_errors = list`?

You are using thread_errors.append above and defaultdict does not have an append method.

Reply:
I meant to use a dictionary that stores exceptions for all the queries with the query_id as the key. I'm passing `thread_errors[i]` to `run_query` (not `thread_errors`)